### PR TITLE
Closes support Reduce motion preference, #3437

### PR DESCRIPTION
**Pull request by low-batt**
Added AccessiblityPreference struct containing methods that make it
easy for other code to handle the Reduce motion preference.

Modified methods in MainWindowController that support entering and
exiting full screen mode and showing and hiding the side bar to
suppress animations if Reduce motion has been enabled.

- [ ] This change has been discussed with the author.
- [x ] It implements / fixes issue #3437.

---

**Description:**

### CODE REVIEW

File: iina/AccessibilityPreference.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+//  AccessibilityPreference.swift
+//  iina
+//
+//  Created by low-batt on 6/26/21.
+//  Copyright © 2021 lhc. All rights reserved.
+//
+
+import Foundation
+
+struct AccessibilityPreference {
```
Comment (saagarjha) [inline iina/AccessibilityPreference.swift:11]:
```suggestion
struct AccessibilityPreferences {
```

File: iina/AccessibilityPreference.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  /// Adjusts an animation to be instantaneous if the macOS System Preference Reduce motion is enabled.
+  /// - Parameter duration: Desired animation duration.
+  /// - Returns: `0` if reduce motion is enabled; otherwise the given duration.
+  static func adjustDuration(_ duration: TimeInterval) -> TimeInterval {
```
Comment (saagarjha) [inline iina/AccessibilityPreference.swift:16]:
```suggestion
  static func adjustedDuration(_ duration: TimeInterval) -> TimeInterval {
```

File: iina/AccessibilityPreference.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  /// does not need to concern itself with this preference not being available until macOS Sierra.
+  ///
+  /// Proper handling of the Reduce motion preference setting is covered in the
+  /// [Apple Human Interface Guidelines under Appearance Effects and Motion](https://developer.apple.com/design/human-interface-guidelines/accessibility/overview/appearance-effects/).
+  ///
+  /// To change this preference, choose Apple menu > System Preferences, click Accessibility, click Display, then click Display and
+  /// check or uncheck Reduce motion.
+  ///
+  /// - Returns: `true` if reduce motion is enabled; otherwise `false`.
+  static func reduceMotion() -> Bool {
```
Comment (saagarjha) [inline iina/AccessibilityPreference.swift:32]:
```suggestion
  static var motionReductionEnabled: Bool {
```

**Review by saagarjha — COMMENTED** [review]
Accessibility is important; thanks for making these changes. A couple comments, though.

File: iina/AccessibilityPreference.swift (hunk 1 above)
Comment (low-batt) [reply iina/AccessibilityPreference.swift]: Fixed.

File: iina/AccessibilityPreference.swift (hunk 2 above)
Comment (low-batt) [reply iina/AccessibilityPreference.swift]: Fixed.

File: iina/AccessibilityPreference.swift (hunk 3 above)
Comment (low-batt) [reply iina/AccessibilityPreference.swift]: Opps. Messed this one up. Missed the bit about switching to computed property. Will drink some coffee and will fix the fix.

Comment (low-batt) [reply iina/AccessibilityPreference.swift]: The 3rd commit addresses this review comment properly. Let me know if I need to be rebasing or squashing.

**Pull request merged**